app/services/publish_on_behalf_interactor.rb

```diff
@@ -0,0 +1,64 @@
+require 'services/base_interactor'
+
+# Publishes a resource on behalf of an organization
+class PublishOnBehalfInteractor < BaseInteractor
+  attr_reader :builder_envelope, :builder_errors
+
+  def call(params)
+    organization = Organization.find(params[:organization_id])
+    publisher = params[:current_user].publisher
+
+    organization_publisher = fetch_or_create_organization_publisher(organization, publisher)
+
+    unless organization_publisher
+      @error = [
+        'User\'s publisher is not authorized to publish on behalf of this organization',
+        401
+      ]
+      return
+    end
+
+    @builder_envelope, @builder_errors = EnvelopeBuilder.new(
+      envelope_attributes(params.merge(organization_publisher: organization_publisher))
+    ).build
+  end
+
+  private
+
+  def fetch_or_create_organization_publisher(organization, publisher)
+    organization_publisher = OrganizationPublisher
+                             .where(organization: organization)
+                             .where(publisher: publisher)
+                             .first
+
+    return organization_publisher if organization_publisher
+
+    return nil unless publisher.super_publisher
```

Comment:
maybe some comments on these rules, just to be clear for anyone who gets to this code later.